### PR TITLE
CCS-14: Create Route Handler for Delete Functionality

### PR DESCRIPTION
**Pull request by jmusone**
Added delete clothing handler.

### CODE REVIEW

File: InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/DeleteClothingHandler.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    private final ClothingService service;
+
+    public DeleteClothingHandler(ClothingService service) {
+        this.service = service;
+    }
+
+    @Override
+    public Mono<ServerResponse> handle(ServerRequest serverRequest) {
+        return serverRequest.bodyToMono(UpdateClothingHandler.Request.class)
+                .flatMap(new Function<UpdateClothingHandler.Request, Mono<? extends Clothing>>() {
```
Comment (hitchan) [inline InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/DeleteClothingHandler.java]: You don't need a cast here, you can use a lambda.

File: InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/DeleteClothingHandler.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                                .condition(request.getCondition())
+                                .status(Item.Status.ISSUED)
+                                .brand(request.getBrand())
+                                .color(request.getColor())
+                                .type(request.getType())
+                                .gender(request.getGender())
+                                .size(request.getSize())
+                                .build());
+                    }
+                }).flatMap(new Function<Clothing, Mono<? extends ServerResponse>>() {
```
Comment (hitchan) [inline InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/DeleteClothingHandler.java]: Same here

File: InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/DeleteClothingHandler.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                    @Override
+                    public Mono<? extends ServerResponse> apply(Clothing clothing) {
+                        return ServerResponse.ok()
+                                .body(BodyInserters.fromValue(UpdateClothingHandler.Response.from(clothing)));
+                    }
+                })
+                .onErrorResume(ex -> ServerResponse.badRequest().build());
```
Comment (hitchan) [inline InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/DeleteClothingHandler.java]: Bad request, or 4xx errors are typically used for client side problems and suggests the client to modify their request.
If the problem is coming internally, these should be 5xx style errors.

File: InventoryManagement/src/main/java/com/ccs/inventorymanagement/service/ClothingService.java [hunk 4]
```diff
@@ -44,7 +44,11 @@ public Mono<Clothing> update(Clothing item) {
...
 
     @Override
-    public Mono<Clothing> delete(UUID id) {
-        return null;
+    public Mono<Clothing> delete(Clothing item) {
+        return clothingRepository.save(ClothingEntity.from(item)
+                        .toBuilder()
+                        .updated(Instant.now())
+                        .build())
+                .map(Clothing::from);
```
Comment (hitchan) [inline InventoryManagement/src/main/java/com/ccs/inventorymanagement/service/ClothingService.java]: A delete call should not be passing you an entity to save, you should just receive the ID of the entity requesting to be removed and then looking it up, and if it exists, perform your delete logic.

**Review by hitchan — CHANGES_REQUESTED** [review]
Please review the comments and let me know what you think

File: InventoryManagement/src/main/java/com/ccs/inventorymanagement/repo/ClothingRepository.java [hunk 5]
```diff
@@ -1,10 +1,14 @@
...
 import org.springframework.data.r2dbc.repository.R2dbcRepository;
 import org.springframework.stereotype.Repository;
+import reactor.core.publisher.Mono;
 
 import java.util.UUID;
 
 @Repository
 public interface ClothingRepository extends R2dbcRepository<ClothingEntity, UUID> {
+    @Query("UPDATE clothing SET status = 'ISSUED' WHERE id = $0")
+    Mono<Void> deleteByID(UUID id);
```
Comment (hitchan) [inline InventoryManagement/src/main/java/com/ccs/inventorymanagement/repo/ClothingRepository.java]: You may want to rename this method to let callers know what it really does. Something like "softDelete()" or something like that. 99.9% of the time "delete" in a repo means to remove the row completely. You could even make an update method that takes the enum so you could implement undo operations.

File: InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/DeleteClothingHandler.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    private final ClothingService service;
+
+    public DeleteClothingHandler(ClothingService service) {
+        this.service = service;
+    }
+
+    @Override
+    public Mono<ServerResponse> handle(ServerRequest serverRequest) {
+        return serverRequest.bodyToMono(UpdateClothingHandler.Request.class)
+                .flatMap((Function<UpdateClothingHandler.Request, Mono<? extends Clothing>>) request -> service.update(Clothing.builder()
```
Comment (hitchan) [inline InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/DeleteClothingHandler.java]: Remove the cast, it is not required.

File: InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/DeleteClothingHandler.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+                        .name(request.getName())
+                        .description(request.getDescription())
+                        .condition(request.getCondition())
+                        .status(Item.Status.ISSUED)
+                        .brand(request.getBrand())
+                        .color(request.getColor())
+                        .type(request.getType())
+                        .gender(request.getGender())
+                        .size(request.getSize())
+                        .build())).flatMap((Function<Clothing, Mono<? extends ServerResponse>>) clothing -> ServerResponse.ok()
```
Comment (hitchan) [inline InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/DeleteClothingHandler.java]: Same here

**Review by hitchan — CHANGES_REQUESTED** [review]

File: InventoryManagement/src/main/java/com/ccs/inventorymanagement/repo/ClothingRepository.java [hunk 8]
```diff
@@ -1,10 +1,14 @@
...
+import org.springframework.data.r2dbc.repository.Query;
 import org.springframework.data.r2dbc.repository.R2dbcRepository;
 import org.springframework.stereotype.Repository;
+import reactor.core.publisher.Mono;
 
 import java.util.UUID;
 
 @Repository
 public interface ClothingRepository extends R2dbcRepository<ClothingEntity, UUID> {
+    @Query("UPDATE clothing SET status = 'ISSUED' WHERE id = $0")
```
Comment (hitchan) [inline InventoryManagement/src/main/java/com/ccs/inventorymanagement/repo/ClothingRepository.java]: Does ISSUED mean deleted in this system?

**Review by hitchan — APPROVED** [review]

**Review by RRDarius — APPROVED** [review]

**Pull request merged**